backend/pkg/api/router.go

```diff
@@ -103,11 +103,12 @@ func addRoutes(hs *handlers.HandlerService, publicRouter, internalRouter *mux.Ro
 
 		{http.MethodPost, "/users", nil, hs.InternalPostUsers},
 		{http.MethodPost, "/users/email-confirmations/{token}", nil, hs.InternalPostUserConfirm},
+		{http.MethodPut, "/users/password/reset", nil, hs.InternalPutUserPasswordReset},
```

Comment:
The reasons I've chosen `PUT` here are:
- most examples you can find online around this suggest using it
- the endpoints should be idempotent: no matter how often you call them, the result should be the same. This is the case here imo
- it further distinguishes the endpoint from the `POST` version
- we're also using it for email change requests (`/users/me/email`)

So I'd keep it, if you've a strong opinion about it pls lmk why.

Reply:
Generally, `PUT` is used to update/replace an existing resource. Although it can be used to create new resources, this is rather an exception and we currently only use it for replacing existing resources in our API. I would argue that this endpoint creates a new password reset request (and does not replace an existing one), so from a consistency perspective it should be `POST`. 
But, more importantly: You are correct that `PUT` endpoints must be idempotent, which is not the case for this endpoint. Calling the endpoint twice will either return a forbidden error (due to the password reset ts) or set a new, distinct password reset token for the user. Both of these are not the same result as the first call. If you try to call `PUT /users/me/email` or even `PUT /users/me/password` with the same parameters, it will always result in the same DB state.